integrationtests/online_dataset/test_online_dataset.py

```diff
@@ -339,6 +342,47 @@ def test_dataset_impl(
         if image_bytes not in FIRST_ADDED_IMAGES:
             raise ValueError(f"Could not find image {idx} in created images, all_samples = {all_samples}")
 
+    if not consistency_check:
+        return
+
+    print("Iterating again to check across epochs.")
+
+    second_samples = []
+    second_data = []
+    second_labels = []
+
+    for batch_number, batch in enumerate(dataloader):
+        sample_ids = batch[0]
+        if isinstance(sample_ids, torch.Tensor):
+            sample_ids = sample_ids.tolist()
+        elif isinstance(sample_ids, tuple):
+            sample_ids = list(sample_ids)
+
+        assert isinstance(sample_ids, list), "Cannot parse result from DataLoader"
+        assert isinstance(batch[1], torch.Tensor) and isinstance(batch[2], torch.Tensor)
+
+        second_samples.extend(sample_ids)
+        for sample in batch[1]:
+            second_data.append(sample)  # iterate over batch dimension to extract samples
+        second_labels.extend(batch[2].tolist())
+
+    # Same content, but not same order
+    # (even without shuffle, the storage may return samples in a slightly different order)
+
+    assert set(second_samples) == set(
+        all_samples
+    ), f"second_samples = {second_samples} \n\n all_samples = {all_samples}"
+    assert set(second_labels) == set(all_labels), f"second_labels = {second_labels} \n\n all_labels = {all_labels}"
+    for data1 in second_data:
+        assert any(torch.allclose(data1, data2) for data2 in all_data)
+
+    # when shuffling, we expect a different order
+
+    if shuffle:
+        assert second_samples != all_samples, f"second_samples = {second_samples} \n\n all_samples = {all_samples}"
+        assert not all(torch.allclose(data1, data2) for data1, data2 in zip(second_data, all_data))
+        assert second_labels != all_labels, f"second_labels = {second_labels} \n\n all_labels = {all_labels}"
+
```

Comment:
I like the supplemented tests here. Thanks

Reply:
thank you :)